src/main/java/mate/academy/webshop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,127 @@
+            createdProduct = list.get(list.size() - 1);
+        } catch (SQLException e) {
+            LOGGER.error("Can't create product", e);
+            return null;
+        }
+        return createdProduct;
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * FROM products WHERE id = ?";
+        Product product;
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            product = getProductFromResultSet(statement.executeQuery());
+        } catch (SQLException e) {
+            LOGGER.error(String.format("Can't get product with id: %d!", id), e);
+            return Optional.empty();
+        }
+        return Optional.of(product);
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products";
+        List<Product> products;
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            products = getAllProductsFromResultSet(statement.executeQuery());
+        } catch (SQLException e) {
+            LOGGER.error("Can't get list of products!", e);
+            return new ArrayList<>();
+        }
+        return products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        String query = "UPDATE products SET name = ?, price = ? WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, product.getName());
+            statement.setDouble(2, product.getPrice());
+            statement.setLong(3, product.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            LOGGER.error(String.format("Can't update product with id: %d!",
+                    product.getId()), e);
+            return null;
+        }
+        return product;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "DELETE FROM products WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            LOGGER.error(String.format("Can't delete product with id: %d!", id));
+            return false;
+        }
+        return true;
+    }
+
+    public Product getProductFromResultSet(ResultSet resultSet)
+            throws SQLException {
+        Product product;
+
+        long productId = resultSet.getLong("id");
+        String name = resultSet.getString("name");
+        double price = resultSet.getDouble("price");
+        product = new Product(name, price);
```

Comment:
```suggestion Product product = new Product(name, price); ``` 
declare a variable as close as possible to their first usage

Reply:
Got it. Optimized.